pkg/measure-config/src/measurefs.go

```diff
@@ -0,0 +1,278 @@
+		}
+	}
+	return false
+}
+
+// theses file may appear later on the device and we record the
+// fact that file exists. during attestation process we can detect
+// this fact by comparing saved and current event log
+func getDangerousList() []string {
+	return []string{
+		"/config/bootstrap-config.pb",
+		"/config/GlobalConfig/override.json",
+		"/config/GlobalConfig/global.json",
+		"/config/Force-API-V1",
+	}
+}
+
+func sha256sumForFile(filePath string) (string, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hash := sha256.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+func sha256sumString(s string) [32]byte {
+	return sha256.Sum256([]byte(s))
+}
+
+func measureFileContent(filePath string, tpm io.ReadWriter, tpmPassword string) (*TpmEvent, error) {
+	hash, err := sha256sumForFile(filePath)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot measure %s :%v", filePath, err)
+	}
+
+	event_data := fmt.Sprintf("file:%s hash:%s", filePath, hash)
+
+	// it seems PCRExtend expects a hash not data itself.
+	event_data_hash := sha256sumString(event_data)
+
+	err = tpm2.PCRExtend(tpm, ConfigPCRHandle, tpm2.AlgSHA256, event_data_hash[:], tpmPassword)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot measure %s. couldn't extend PCR: %v", filePath, err)
+	}
+
+	pcr, err := readConfigPCR(tpm)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot measure %s. couldn't read PCR: %v", filePath, err)
+	}
+
+	return &TpmEvent{event_data, pcr}, nil
+}
+
+func measureFilePath(filePath string, tpm io.ReadWriter, tpmPassword string, exist bool) (*TpmEvent, error) {
+	event_data := fmt.Sprintf("file:%s exist:%t", filePath, exist)
+	// it seems PCRExtend expects a hash not data itself.
+	event_data_hash := sha256sumString(event_data)
+
+	err := tpm2.PCRExtend(tpm, ConfigPCRHandle, tpm2.AlgSHA256, event_data_hash[:], tpmPassword)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot measure path %s. couldn't extend PCR: %v", filePath, err)
+	}
+
+	pcr, err := readConfigPCR(tpm)
+
+	if err != nil {
+		return nil, fmt.Errorf("cannot measure path %s. couldn't read PCR: %v", filePath, err)
+	}
+
+	return &TpmEvent{event_data, pcr}, nil
+}
+
+func getTPMCredentials() (string, error) {
+	// password, err := evetpm.ReadOwnerCrdl()
+
+	// // on the first run /config/tpm_credential may not exist
+	// // but this is not a problem. In this case TPM is accessible
+	// // without a password
+	// if os.IsNotExist(err) {
+	// 	return "", nil
+	// }
+
+	// // for any other error just return it
+	// if err != nil {
+	// 	return "", err
+	// }
+	// return password, nil
+
+	// now we measure config just after UEFI has finished its part
+	// the session is not encrypted until pillar is started and we
+	// must not use the password from tpm_credential
+	// but it may be changed in future if the code is moved to other
+	// application
```

Comment:
If so shouldn't this comment be before the function definition?

If it is useful to keep the above commented code around it might be better to put it in an if false branch instead of being comments.

Reply:
fixed